monitor/statistics.hpp

```diff
@@ -28,6 +28,10 @@
 #include <mutex>
 #include <sstream>
 #include <thread>
+#include <map>
+
+#include <boost/thread/locks.hpp>
+#include <boost/thread/shared_mutex.hpp>
```

Comment:
The only real places where boost is needed are tests and python.
My initial implementation used boost intrusive lists, but I thought it was changed.

Anyway, do not add new dependencies, it would be great to get rid of current use cases, they are quite trivial like boost::lexical_cast (we have much faster helper) and the like